Source/HTTPHeaders.swift

```diff
@@ -24,51 +24,215 @@
 
 import Foundation
 
-public typealias HTTPHeaders = [String: String]
+public struct HTTPHeaders {
+    private var headers = [HTTPHeader]()
+    
+    public init() { }
+    
+    public init(_ headers: [HTTPHeader]) {
```

Comment:
This initializer is the only place where it is possible to insert headers with duplicate names. All other initializers and mutating functions perform deduplication.

Reply:
Yep, changed.